### ISSUE TITLE
[RFE] Support for starting scan with localStorage instead of cookie

### ISSUE THREAD

**Issue by brandonprry** [Arachni/arachni#603]
In the new age of javascript-only web applications, there is sometimes a need to get through an agegate or similar gating mechanism that sets information into localStorage, which in turn allows access to the rest of the javascript application.

It would be great if a mechanism existed to kick off a scan with localStorage values, similar to kicking off a scan with a predefined cookie.

Thanks!

**Comment by cclements** [comment]
I second this.

**Comment by Zapotek** [comment]
I should be able to swing it by adding hooks for plugins, like running custom JS prior to each page being loaded.
You'd still need to write a custom plugin for it though, not sure how clean it would be if done via the CLI.

**Comment by brandonprry** [comment]
The way I imagine it working is pointing arachni at a JSON file which defines the key/value pairs that are inserted into localStorage before the scan starts, but it might be a bit naive.

**Comment by Zapotek** [comment]
No it's actually a good idea.

**Comment by brandonprry** [comment]
This is great dude, I will test this out next week. :)

**Comment by Zapotek** [comment]
Sorry it took so long, been working on some really nice optimizations for the version after next and some other stuff. :)
Pushing nightlies now.

**Comment by cclements** [comment]
Can I get a little more documentation on how to use this particular feature?  I've got an angularjs app that uses localstorage values for authentication, and I *think* I've constructed my json file correctly for arachni to use, but it doesn't seem to work during scanning.  I'm wondering how to set what I see in my browsers dev tools into what json format arachni needs to use it correctly.

For clarity, my dev tools show my localstorage as:

access_token - "somestring"
debug - undefined
profile - {"name":"somename"(truncated)}
token - "someotherstring"

and I have created a json file for arachni:
{
    "access_token": "somestring",
    "debug": "undefined",
    "profile": {"name":"somename"(truncated)},
    "token": "someotherstring"
}
